compiler/rustc_session/src/config.rs

```diff
@@ -1005,6 +1006,96 @@ pub fn to_crate_config(cfg: FxHashSet<(String, Option<String>)>) -> CrateConfig
     cfg.into_iter().map(|(a, b)| (Symbol::intern(&a), b.map(|b| Symbol::intern(&b)))).collect()
 }
 
+/// The parsed `--check-cfg` options
+pub struct CheckCfg<T = String> {
+    /// Set if `names()` checking is enabled
+    pub names_checked: bool,
+    /// The union of all `names()`
+    pub names_valid: FxHashSet<T>,
+    /// The set of names for which `values()` was used
+    pub values_checked: FxHashSet<T>,
+    /// The set of all (name, value) pairs passed in `values()`
+    pub values_valid: FxHashSet<(T, T)>,
+}
+
+impl<T> Default for CheckCfg<T> {
+    fn default() -> Self {
+        CheckCfg {
+            names_checked: false,
+            names_valid: FxHashSet::default(),
+            values_checked: FxHashSet::default(),
+            values_valid: FxHashSet::default(),
+        }
+    }
+}
+
+impl<T> CheckCfg<T> {
+    fn map_data<O, F>(&self, f: F) -> CheckCfg<O>
+    where
+        F: Fn(&T) -> O,
+        O: std::cmp::Eq + std::hash::Hash,
+    {
+        CheckCfg::<O> {
+            names_checked: self.names_checked,
+            names_valid: (&self.names_valid).into_iter().map(|a| f(a)).collect(),
+            values_checked: (&self.values_checked).into_iter().map(|a| f(a)).collect(),
+            values_valid: (&self.values_valid).into_iter().map(|(a, b)| (f(a), f(b))).collect(),
```

Comment:
```suggestion
            values_valid: self.values_valid.iter().map(|(a, b)| (f(a), f(b))).collect(),
```
?

Same in lines above.